src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,222 @@
+        super(DDM, self).__init__(configDict)
+        self['logger'].debug("Initializing DDM with url: %s", self['endpoint'])
+
+    def _getResult(self, ddmReq, apiUrl, callname=""):
+
+        cachedApi = "%s.json" % callname
+
+        # striping the None values from the request before sending it to dynamo
+        ddmReq.strip()
+
+        # The Request class expects this to be a dict, otherwise it tries to
+        # encode the inputdata once again and inserts parasite information and
+        # breaks the call to dynamo.
+        ddmReq = dict(ddmReq)
+        data = self.refreshCache(cachedApi,
+                                 apiUrl,
+                                 inputdata=ddmReq,
+                                 verb='POST')
+        results = data.read()
+        data.close()
+
+        results = json.loads(results)
+        return results
+
+    def makeRequest(self, ddmReq, api=None):
+        """
+        _makeRequest_
+
+        Makes a SINGLE request to DDM
+
+        :ddmReq: Request template an instance of DDMReqTemplate class.
+        :api:    The DDM api to be queried.
+        :return: Dictionary containing the result from DDM
+        """
+
+        # try to figure out the api from the ddmReq template
+        if not api:
+            api = ddmReq.api
+
+        apiUrl = "registry/request/{}".format(api)
+        callname = "DDM-{}".format(api)
+        # TODO: here to send to the logger the rquest parameters
+        return self._getResult(ddmReq, apiUrl, callname)
+
+    def makeAggrRequests(self, ddmReqList, aggrKey=None):
+        """
+        _makeAggrRequest_
+
+        Makes several Aggregated requests to DDM. Takes the list of objects,
+        instances of ddmReqTemplate and aggregates it based on a selected key.
+        Uses the ddmReqTemplate internal Equivalence method for comparison with
+        the aggregation key excluded. The final result should be a list of unique
+        ddmRequests each one of them containing all equal ddmRequests from the
+        initial list with the values of the aggregation key accumulated in a list
+        under the same key. Creates a request per ddmReqTemplate from the list
+        with aggregated templates.
+        NOTE:
+            In this method the api is always fetched from the template.
+        TODO:
+            So far it is working basically for the 'copy' API and due to that
+            the aggregation key is always a dictionary so no explicit check is
+            done for the type. In case we decide to use it we should develop
+            this to the end with the proper type checks.
+
+        :ddmReqList: List of objects, instances of DDMReqTemplate class.
+        :aggrKey:    String representing the Aggregation key.
+        :return:     List of Dictionaries containing the results from DDM
+
+        """
+        # TODO: to put a check for aggrKey == None
+        ddmReqListAgg = []
```

Comment:
how about starting this variable name without `ddm`? It's very easy to get lost in the block below :)

Reply:
I will try to do something about it.

Reply:
Done + few more variable name changes.

Reply:
I also changed the cycle. It was not behaving as expected. Now it is doing the job right.